src/Communication/Connection.php

```diff
@@ -106,6 +106,30 @@ public function __construct($socketClient, LoggerInterface $logger = null, int $
 
         // create socket client
         if (\is_string($socketClient)) {
+
+            if (\in_array(\parse_url($socketClient, PHP_URL_SCHEME), ['http', 'https'])) {
+
+                $configURL = $socketClient.'/json/version';
+
+                $resp = \file_get_contents($configURL);
```

Comment:
We can't call this without doing any error handling (yes, I really do mean "error") in the PHP 5 sense of the word.